src/Rules/DeadCode/PossiblyPureStaticCallCollector.php

```diff
@@ -0,0 +1,56 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\DeadCode;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Expression;
+use PHPStan\Analyser\Scope;
+use PHPStan\Collectors\Collector;
+use function count;
+use function strtolower;
+
+/**
+ * @implements Collector<Node\Stmt\Expression, array{class-string, string, int}>
+ */
+class PossiblyPureStaticCallCollector implements Collector
+{
+
+	public function __construct()
+	{
+	}
+
+	public function getNodeType(): string
+	{
+		return Expression::class;
+	}
+
+	public function processNode(Node $node, Scope $scope)
+	{
+		if (!$node->expr instanceof Node\Expr\StaticCall) {
+			return null;
+		}
+		if (!$node->expr->name instanceof Node\Identifier) {
+			return null;
+		}
+
+		if ($node->expr->class instanceof Node\Name) {
+			$calledOnType = $scope->resolveTypeByName($node->expr->class);
+		} else {
+			$calledOnType = $scope->getType($node->expr->class);
```

Comment:
We shouldn't report anything if it's Expr. Keep it simple. We'd have to deal with the polymorphism again.